Walk and build wheels for every integration

trishankatdatadog opened the pull request:
### What does this PR do?

Walks over each integration, and tries to build wheels for it.

### Motivation

Automatic building and distribution of integrations.

### Testing Guidelines

N/A.

### Versioning

N/A.

### Additional Notes

TBD.

truthbk commented on activemq_xml/setup.py:
```diff
@@ -4,7 +4,10 @@
 
-import simplejson as json
+try:
+    import json
```
I _think_ we want this the other way around: `simplejson` would be the ideal but optional lib, while `json` should be readily available. 😅

trishankatdatadog replied:
From a security point of view, I think a good argument could be made to prefer using `json` (1st-party) instead of `simplejson` (3rd-party). What do you think?

truthbk replied:
That's fine with me, I get that.

So the thing about this snippet is that we'll never really get an `ImportError` on `import json` as it's part of the std library. That's why I suggested the reverse:
```
try:
    import simplejson as json
except ImportError:
    import json
``` 
If `simplejson` is available, use that, otherwise fallback to the standard json library.

Since we still don't seem to want that, then I guess you can just remove the import for `simplejson` altogether.

truthbk commented on make-wheels.py:
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
```
I don't personally have any major objections to this, but I'm wondering if maybe we should keep the project pure python2 (for a better developer experience) for now.

truthbk commented on make-wheels.py:
```diff
@@ -0,0 +1,40 @@
+
+def main():
+    # Get non-hidden directories.
+    dirs = sorted(name for name in os.listdir('.') \
```
you might be able to get every dir you need with something like:
```
dirs = [os.path.dirname(d) for d in glob.glob('./**/setup.py')]
```
That could maybe simplify the logic in the for loop a little bit. This is fine, just a thought.

truthbk commented on make-wheels.py:
```diff
@@ -0,0 +1,40 @@
+            print('Trying to build {}...'.format(name), end='')
+            # https://packaging.python.org/tutorials/distributing-packages/#wheels
+            cmd = 'python setup.py bdist_wheel'.split()
+            proc = subprocess.run(cmd, check=True, stdout=subprocess.PIPE)
```
I think we probably want to capture stderr as well.

trishankatdatadog commented:
Incorporated all of your changes [except for using simplejson by default] :)

On Wed, Nov 29, 2017 at 4:51 AM, Jaime Fullaondo <notifications@github.com>
wrote:

> *@truthbk* commented on this pull request.
>
> This looks like it's almost ready to go. Added a couple of notes, most
> importantly the python3 v python2 question regarding the builder script.
> ------------------------------
>
> In activemq_xml/setup.py
> <https://github.com/DataDog/integrations-core/pull/911#discussion_r153733887>
> :
>
> > @@ -4,7 +4,10 @@
>  from codecs import open
>  from os import path
>
> -import simplejson as json
> +try:
> +    import json
>
> I *think* we want this the other way around: simplejson would be the
> ideal but optional lib, while json should be readily available. 😅
> ------------------------------
>
> In make-wheels.py
> <https://github.com/DataDog/integrations-core/pull/911#discussion_r153733918>
> :
>
> > @@ -0,0 +1,40 @@
> +#!/usr/bin/env python3
>
> I don't personally have any major objections to this, but I'm wondering if
> maybe we should keep the project pure python2 (for a better developer
> experience) for now.
> ------------------------------
>
> In make-wheels.py
> <https://github.com/DataDog/integrations-core/pull/911#discussion_r153736367>
> :
>
> > +
> +
> +'''
> +Author: Trishank Karthik Kuppusamy <trishank.kuppusamy@datadoghq.com>
> +Date: Nov 27 2017
> +Description: A simple Python script to build and test wheels.
> +'''
> +
> +
> +import os
> +import subprocess
> +
> +
> +def main():
> +    # Get non-hidden directories.
> +    dirs = sorted(name for name in os.listdir('.') \
>
> you might be able to get every dir you need with something like:
>
> dirs = [os.path.dirname(d) for d in glob.glob('./**/setup.py')]
>
> That could maybe simplify the logic in the for loop a little bit. This is
> fine, just a thought.
> ------------------------------
>
> In make-wheels.py
> <https://github.com/DataDog/integrations-core/pull/911#discussion_r153737151>
> :
>
> > +                  if os.path.isdir(name) \
> +                  and not name.startswith('.'))
> +    # Get the current working directory.
> +    pwd = os.getcwd()
> +    # For each directory:
> +    for name in dirs:
> +        # Get the next working directory.
> +        nwd = os.path.join(pwd, name)
> +        # Change to the next working directory.
> +        os.chdir(nwd)
> +        # If the next working directory has 'setup.py', then we can build.
> +        if os.path.isfile('setup.py'):
> +            print('Trying to build {}...'.format(name), end='')
> +            # https://packaging.python.org/tutorials/distributing-packages/#wheels
> +            cmd = 'python setup.py bdist_wheel'.split()
> +            proc = subprocess.run(cmd, check=True, stdout=subprocess.PIPE)
>
> I think we probably want to capture stderr as well.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/DataDog/integrations-core/pull/911#pullrequestreview-79783109>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AfmSER2Myg8QT0x4dARC8OitIsoW4P93ks5s7SkegaJpZM4QsYEo>
> .
>

masci commented on make-wheels.py:
```diff
@@ -0,0 +1,52 @@
+'''
+Author: Trishank Karthik Kuppusamy <trishank.kuppusamy@datadoghq.com>
+Date: Nov 27 2017
+Description: A simple Python script to build and test wheels.
```
Can we change this header to this one?
```
# (C) Datadog, Inc. 2017
# All rights reserved
# Licensed under Simplified BSD License (see LICENSE)
```

trishankatdatadog replied:
Done!

truthbk commented:
The thing is we'll never really get an `ImportError` on `import json` as it's part of the std library. That's why I suggested the reverse:
```
try:
    import simplejson as json
except ImportError:
    import json
``` 
If `simplejson` is available, use that, otherwise fallback to the standard json library.

If you still don't want that, then I guess you can just remove the import for `simplejson` altogether. 

Also, I rebased `jaime/wheels` to the latest `master` a couple of days ago (maybe yesterday), so we'll need to rebase this one to the latest `jaime/wheels`.

trishankatdatadog commented:
Okay, it wasn't the prettiest merge, but done

trishankatdatadog commented:
If you guys prefer, I can close this, and send a clean PR instead.
